src/codegen/codegen_c_visitor.cpp

```diff
@@ -69,13 +69,8 @@ void CodegenCVisitor::visit_integer(const Integer& node) {
     if (!codegen) {
         return;
     }
-    const auto& macro = node.get_macro();
     const auto& value = node.get_value();
-    if (macro) {
-        macro->accept(*this);
-    } else {
-        printer->add_text(std::to_string(value));
-    }
+    printer->add_text(std::to_string(value));
 }
```

Comment:
Was there any bug here? 🤔 

Note that the name was printed **_intentionally_**. When we have code like:

```python
DEFINE init 1
DEFINE rise 2
DEFINE fall 3
DEFINE off 4

NET_RECEIVE(w) {
  if (flag == init) {
    WATCH (v > -55) rise
  }else if (flag == rise) {
    g = gna
    e = ena
    WATCH (v > 10) fall
  }else if (flag == fall) {
    g = gk
    e = ek
    WATCH (v < -70) off
  }else if (flag == off) {
    g = gpas
    e = erev
    WATCH (v > -55) rise
  }
}
```
Having these conceptual names like `init`, `fall`, `rise` (of voltage) is useful in the generated code rather than 1/2/3/4. Because we still look at the code during debugging.

Reply:
The problem is that the macro was not defined inside the c++ file.
I will look on this side now.
This PR is only here to "discuss", don't worry ^^

Reply:
I see that you mentioned offending model + mod file. I ran test with WATCH statement
and see that what I mentioned above (as expectation) is not true i.e. DEFINE is being replaced with values in same cases. Before merging this PR, I would just verify how in certain cases we are currently replacing DEFINE with values without this PR.

Reply:
We looked with @iomaganaris , and we found that for `BynaryExpression` special case, the integer is converted as parsing to a `Double` so a direct value and no more `macro`.
In the case found in `nmodldb` this is not an expression but a direct value, so we keep the meaning of `integer` that can be `direct value` or `named value`.